tests/upgrade/test_upgrade.py

```diff
@@ -46,15 +67,52 @@ def test_check_storage_integrity(strat_proxy, vault_proxy, deployer, proxy_admin
     ## Check Integrity
     assert old_want == strat_proxy.want()
     assert old_vault == strat_proxy.vault()
-    assert old_withdrawalMaxDeviationThreshold == strat_proxy.withdrawalMaxDeviationThreshold()
+    assert (
+        old_withdrawalMaxDeviationThreshold
+        == strat_proxy.withdrawalMaxDeviationThreshold()
+    )
     assert old_autoCompoundRatio == strat_proxy.autoCompoundRatio()
     assert old_withdrawalSafetyCheck == strat_proxy.withdrawalSafetyCheck()
     assert old_processLocksOnReinvest == strat_proxy.processLocksOnReinvest()
     assert old_bribesProcessor == strat_proxy.bribesProcessor()
     assert old_auraBalToBalEthBptMinOutBps == strat_proxy.auraBalToBalEthBptMinOutBps()
 
-    gov = accounts.at(vault_proxy.governance(), force=True)
+    ## === Test claiming the current round of bribes === ##
+    intial_balance = usdc.balanceOf(strat_proxy.bribesProcessor())
+
+    with open("tests/upgrade/data/hh_claim_data_7_23.json", "r") as read_file:
+        data = json.load(read_file)["data"]
+
+    ## Block taken from badger-multisig script
+    ## Reference: https://github.com/Badger-Finance/badger-multisig/blob/0a98cb268661b63221195eb4683cfadfa05e0120/great_ape_safe/ape_api/badger.py#L164
+    def transform_claimable(amount_string, n_decimals):
+        # if the last number is a zero it gets omitted by the api,
+        # here we pad the matissa with zeroes to correct for this
+        assert "." in amount_string
+        splitted = amount_string.split(".")
+        return splitted[0] + splitted[-1].ljust(n_decimals, "0")
+
+    aggregate = {"tokens": [], "amounts": []}
+    for item in data:
+        aggregate["tokens"].append(item["token"])
+        aggregate["amounts"].append(
+            transform_claimable(item["claimable"], item["decimals"])
+        )
+
+    metadata = [
+        (
+            item["claimMetadata"]["identifier"],
+            item["claimMetadata"]["account"],
+            item["claimMetadata"]["amount"],
+            item["claimMetadata"]["merkleProof"],
+        )
+        for item in data
+    ]
 
-    ## Let's do a quick earn and harvest as well
-    vault_proxy.earn({"from": gov})
-    strat_proxy.harvest({"from": gov})
\ No newline at end of file
+    strat_proxy.claimBribesFromHiddenHand(
+        NEW_REWARDS_DISTRIBUTOR, metadata, {"from": strategist}
+    )
+    # Confirm that claimable amount was claimed and transferred to the Bribes Processor
```

Comment:
does it make sense to generalize a local function like `def confirmBribeClaim()` to wrap following lines and allow to specify the target token to be confirmed?

Reply:
I thought of making a generalized test but given that this should be the last time we upgrade (it is 2 months away from deprecation) I decided to go through the hardcoded route and just test the current situation (only USDC bribes to claim).